aea/test_tools/test_skill.py

```diff
@@ -0,0 +1,368 @@
+        return is_incoming, target
+
+    @staticmethod
+    def _non_initial_incoming_message_dialogue_reference(
+        dialogue: Dialogue,
+    ) -> Tuple[str, str]:
+        dialogue_reference = (
+            dialogue.dialogue_label.dialogue_reference[0],
+            Dialogues._generate_dialogue_nonce()  # pylint: disable=protected-access
+            if dialogue.dialogue_label.dialogue_reference[1]
+            == Dialogue.UNASSIGNED_DIALOGUE_REFERENCE
+            else dialogue.dialogue_label.dialogue_reference[1],
+        )
+        return dialogue_reference
+
+    def prepare_dialogue(
+        self,
+        dialogues: Dialogues,
+        messages: Tuple[DialogueMessage, ...],
+        counterparty: Address = "counterparty",
+    ) -> Dialogue:
+        """
+        Quickly create a dialogue.
+
+        The 'messages' argument is a tuple of DialogueMessages.
+        For every DialogueMessage (performative, contents, is_incoming, target):
+            - if 'is_incoming' is not provided: for the first message it is assumed False (outgoing),
+            for any other message, it is the opposite of the one preceding it.
+            - if 'target' is not provided: for the first message it is assumed 0,
+            for any other message, it is the index of the message before it in the tuple of messages + 1.
+
+        :param dialogues: a dialogues class
+        :param counterparty: the message_id
+        :param messages: the dialogue_reference
+
+        :return: the created incoming message
+        """
+        incomplete_message_1 = messages[0]
+        performative = incomplete_message_1[0]
+        contents = incomplete_message_1[1]
+        message_id = messages.index(incomplete_message_1) + 1
+        is_incoming, target = self._provide_unspecified_fields(
+            incomplete_message_1, last_incoming=True, message_id=1,
+        )
+
+        if is_incoming:  # messages from the opponent
+            dialogue_reference = dialogues.new_self_initiated_dialogue_reference()
+            message = self.build_incoming_message(
+                message_type=dialogues._message_class,  # pylint: disable=protected-access
+                dialogue_reference=dialogue_reference,
+                message_id=message_id,
+                target=target,
+                performative=performative,
+                to=self.skill.skill_context.agent_address,
+                sender=counterparty,
+                **contents,
+            )
+            dialogue = cast(Dialogue, dialogues.update(message))
+            if dialogue is None:
+                raise AEAEnforceError(
+                    "Cannot update the dialogue with message number {}".format(
+                        message_id
+                    )
+                )
+        else:  # messages from self
+            _, dialogue = dialogues.create(
+                counterparty=counterparty, performative=performative, **contents
+            )
```

Comment:
Can be extracted in appropriate private method; then we can use it below again in the for loop. Diff between create and reply can be easily controlled with flag.

Reply:
I took the extraction part into its own private method.

As I said, the treatment of the first message and the rest could be combined into one for loop, but the way `dialogue_reference` is defined then causes a flake8/mypy error. I did not manage to come up with a way to address that. 

If you want, I can submit that code and you can see if you can resolve the flake8/mypy issue?